File: core/repository/api_surface_engine.py
from __future__ import annotations

from typing import Dict

from core.parsers.api_resolution_engine import resolve_api_surface
from core.parsers.parser_registry import ParserRegistry


def extract_api_surface(text: str, path: str = "") -> Dict[str, object]:
    language = ParserRegistry.detect_language(path=path)
    api = resolve_api_surface(text or "", language, path=path)
    return {
        "routes": api.get("routes", []),
        "rest": api.get("rest", False),
        "graphql": api.get("graphql", False),
        "evidence": api.get("evidence", "parser_api_surface"),
    }

File: src/repository/apiSurfaceEngine.ts
/**
 * Converted from Python: core/repository/api_surface_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { resolveApiSurface } from "../parsers/apiResolutionEngine.js";
import { ParserRegistry } from "../parsers/parserRegistry.js";

export function extractApiSurface(text: any, path: any = ""): any {
  var language: any = ParserRegistry.detect_language(path);
  var api: any = resolveApiSurface(py.or2(text, () => ("")), language, path);
  return {"routes": py.get(api, "routes", []), "rest": py.get(api, "rest", false), "graphql": py.get(api, "graphql", false), "evidence": py.get(api, "evidence", "parser_api_surface")};
}
export { ParserRegistry, resolveApiSurface };
